arch/arc/Kconfig

```diff
@@ -85,13 +85,34 @@ config ARC_FIRQ
 	  If FIRQ is disabled, the handle of interrupts with highest priority
 	  will be same with other interrupts.
 
-config	ARC_STACK_CHECKING
+config 	ARC_HAS_STACK_CHECKING
+	bool "ARC has STACK_CHECKING"
+	default y
+	help
+	  ARC is configured with STACK_CHECKING which is a mechanism for
+	  checking stack accesses and raising an exception when a stack
+	  overflow or underflow is detected.
+
+config  ARC_STACK_CHECKING
+	bool
+	help
+	  Use ARC STACK_CHECKING to do stack protection
+
+config	ARC_STACK_PROTECTION
```

Comment:
Is this additional ARC_STACK_PROTECTION option required? Can't the common HW_STACK_PROTECTION be used to enable either the ARC_STACK_CHECKING or MPU_STACK_GUARD?

Reply:
It's a temp value,  and provides some kind of abstraction.  If directly use HW_STACK_PROTECTION, it means when HW_STACK_PROTECTION is selected, it's needed to select ARC_STACK_CHECKING or MPU_STACK_GUARD in the entry of HW_STACK_PROTECTION, and the coressponing logic may be a little complex.

Reply:
If it is temp, it could be an internal, hidden config?

Reply:
It is hidden, there is no prompt